### PR TITLE
Emit major slice counters in the runtime events

### PR DESCRIPTION
**Pull request by kayceesrk**
This PR emits the counters updated as part of the major slice computation. The related discussion is here: https://github.com/ocaml/ocaml/discussions/13321. `olly` has already been updated to emit these counters in the trace: https://github.com/tarides/runtime_events_tools/pull/46. 

I had originally planned to make this PR after #13320 was merged. However, it looks like it may take longer: https://github.com/ocaml/ocaml/pull/13320#issuecomment-2252321125. The change itself can be easily modified to accommodate future changes in the slice computation.

### CODE REVIEW

File: otherlibs/runtime_events/runtime_events.mli [hunk 1]
```diff
@@ -87,6 +87,16 @@ Live blocks of a Domain's major heap pools.
 (**
 Live blocks of a Domain's major heap large allocations.
 @since 5.1 *)
+| EV_C_MAJOR_HEAP_WORDS
+| EV_C_MAJOR_ALLOCATED_WORDS
```
Comment (gasche) [inline otherlibs/runtime_events/runtime_events.mli:141]: The constructors above all have careful documentation. Shouldn't we try to document those as well, or at least make it explicit why they are not documented?

**Review by gasche — COMMENTED** [review]
This is fine overall. I think it would be nice to try to document a bit the new `runtime_counters` constructors, possibly just to say that they correspond to internal implementation details of the GC related to pacing computation.

File: otherlibs/runtime_events/runtime_events.mli (hunk 1 above)
Comment (kayceesrk) [reply otherlibs/runtime_events/runtime_events.mli:141]: Good point. Let me try to document these.

Comment (tmcgilchrist) [reply otherlibs/runtime_events/runtime_events.mli:141]: That would generate useful documentation on the [Runtime_events module](https://ocaml.org/manual/5.2/api/Runtime_events.html). 
Unfortunately some of the other constructors are undocumented.

**Comment by sadiqj** [comment]
I've added some docs for the constructors in https://github.com/kayceesrk/ocaml/pull/16

One problem with this PR (ocaml/ocaml) though is that `major_collection_slice` can get hammered pretty frequently as we try to do opportunistic major slices in both in the minor GC entry and exit barriers. It is called in the spin wait loop and we had to add a flag to prevent emitting tens of thousands of phase events: https://github.com/ocaml/ocaml/blob/trunk/runtime/major_gc.c#L1643

We probably don't want to emit events when that happens. (do we even want to be updating our work in that case?)

**Comment by kayceesrk** [comment]
> (do we even want to be updating our work in that case?)

Good point. I see that `slice_target` and `slice_budget` are updated in the case of opportunistic slices. We'd have to do something about them.

I'd prefer to keep logic changes in a separate PR. For now, I've updated `update_major_slice_work` to take in `log_events` as an argument. The `caml_gc_log` and `caml_gc_message` remain unguarded as is the case with `major_collection_slice`.

**Review by gasche — APPROVED** [review]
I am happy with the current state. Approved. Thanks @sadiqj for the review, suggestion to use `log_events`, and documentation.

**Comment by kayceesrk** [comment]
Thanks @gasche.

**Pull request merged**